src/test/java/com/wolox/training/controllers/UserControllerTests.java

```diff
@@ -0,0 +1,242 @@
+package com.wolox.training.controllers;
+
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.wolox.training.models.Book;
+import com.wolox.training.models.User;
+import com.wolox.training.repositories.BookRepository;
+import com.wolox.training.repositories.UserRepository;
+import com.wolox.training.support.factories.BookFactory;
+import com.wolox.training.support.factories.UserFactory;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(UserController.class)
+
+public class UserControllerTests {
+
+  @MockBean
+  UserRepository mockUserRepository;
+
+  @MockBean
+  BookRepository mockBookRepository;
+
+  @Autowired
+  MockMvc mvc;
+
+  private User user = new UserFactory().createUserWithoutBooks();
+  List<User> oneUserList = Arrays.asList(user);
+  private User       secondUser = new UserFactory().createUserWithoutBooks();
+  private List<User> allUsers   = Arrays.asList(user, secondUser);
+  private Book       book       = new BookFactory().createBookWithOutUser();
```

Comment:
Just some light here, by convention in Java we don't add spaces to align the assignations. One space is just fine :)